Fix process for invalid serial number entries

John2202W opened the pull request:
Fixed the handling of invalid serial numbers:
- Properly return tuple from user_inputs. Previously it was not returning a tuple for the exception and instead parsed the serial number as "x" i.e. "E**x**ception"[1].
- Properly raises exception now which prevents starting the sequence and returns an error code
- Add info to the UI to display that an exception has occurred (previously if using GUI - would stay at "loading" and issue could only be noticed in the background terminal).

Other:
- Created enum to make return codes clearer
- Prevent unhandled exception caused by trying to close the csv_writer if fixate failed before opening it.

John2202W commented:
Found a bug in the current test_ui_requests that we were currently skipping because of, will fix them up and then add tests for user_serial and other ui requests aswell.

John2202W commented:
Tests now fixed

jcollins1983 reviewed (commented):
Please see comments.

John2202W commented on test/core/test_ui_requests.py:
```diff
@@ -1,34 +1,65 @@
         if target:
             try:
-                ret_val = target(**kwargs)
+                if self.ret_value is None:
```
self.return_value is a special attribute of the magicmock class, it was only being used 'properly' for one test, so considered it better to use a different variable name so as not to mislead it's use. I'll take another look - could be a better way to make it clearer

John2202W commented:
@jcollins1983 I've addressed the comments.

Think there is a bit more improving that can be done. i.e. if the validation against the target fails, no exceptions are raised - it will display the error and continue... The test scripts need to check the return or when they try to use the value thenthings can potentially break. Got to decide whether we leave it up to the test scripts to check the return value or raise an exception in the input validation?
